### PR TITLE
Adopting to context caching

### PR DESCRIPTION
**Pull request by romanovvlad**

### CODE REVIEW

**Comment by romanovvlad** [comment]
The patch adjust several tests to start working with possibly cached context. The caching is being implemented in https://github.com/intel/llvm/pull/2805

File: SYCL/Basic/buffer/buffer_dev_to_dev.cpp [hunk 1]
```diff
@@ -21,21 +21,34 @@ using namespace cl::sycl;
 int main() {
   int Data[10] = {-1, -1, -1, -1, -1, -1, -1, -1, -1, -1};
   {
-    buffer<int, 1> Buffer(Data, range<1>(10),
-                          {property::buffer::use_host_ptr()});
 
-    default_selector Selector;
+    const std::vector<platform> Platforms = platform::get_platforms();
+
+    if (Platforms.size() < 2) {
```
Comment (romanovvlad) [inline SYCL/Basic/buffer/buffer_dev_to_dev.cpp:29]: The test is executed with one of `*_RUN_PLACEHOLDER` set which might result in having only one platform here.
Probably, we should run the test without any `_RUN_PLACEHOLDER` value set.
@vladimirlaz What do you think?

Comment (vladimirlaz) [reply SYCL/Basic/buffer/buffer_dev_to_dev.cpp:29]: we have host device for all cases except acc AFAIR.
If host device does not fit test needs the test can be updated to run on OpenCL BE only

**Review by vladimirlaz — APPROVED** [review]

**Comment by vladimirlaz** [comment]
@smaslov-intel,  could you please review?

File: SYCL/Basic/buffer/buffer_dev_to_dev.cpp [hunk 2]
```diff
@@ -21,21 +23,35 @@ using namespace cl::sycl;
...
+      std::cout << "Need at least two platforms to create two separate "
```
Comment (smaslov-intel) [inline SYCL/Basic/buffer/buffer_dev_to_dev.cpp:30]: This claim does sound true. You can create separate context even with single device (a context to each of its subdevice). What exactly does the "context caching" mean?

File: SYCL/Plugin/enqueue-arg-order-image.cpp [hunk 3]
```diff
@@ -277,7 +277,10 @@ int main() {
 
   remind();
   testcopyD2HImage();
-  testcopyH2DImage();
+  // Disabling this part of the test because there is no reliable way to create
+  // two separate context in all cases which is required to trigger PI calls
```
Comment (smaslov-intel) [inline SYCL/Plugin/enqueue-arg-order-image.cpp:281]: Please clarify why there is no way to test H2D while D2H above is OK. How is this connected to "context caching"?

File: SYCL/Basic/buffer/buffer_dev_to_dev.cpp (hunk 2 above)
Comment (romanovvlad) [reply SYCL/Basic/buffer/buffer_dev_to_dev.cpp:30]: Context caching means that SYCL RT always returns the same context for a given platform.
PR which adds context caching https://github.com/intel/llvm/pull/2805.

>> You can create separate context even with single device (a context to each of its subdevice).

Yes, this is an option as well, but this requires that a device supports splitting to several sub-devices. Probably having such a device is more common than having two platforms on the system though.

**Comment by romanovvlad** [comment]
Superseded by https://github.com/intel/llvm-test-suite/pull/433

**Pull request closed without merging**